src/psyclone/psyir/nodes/assignment.py

```diff
@@ -183,3 +183,19 @@ def reference_accesses(self, var_accesses):
         self.rhs.reference_accesses(var_accesses)
         var_accesses.merge(accesses_left)
         var_accesses.next_location()
+
+    @property
+    def is_array_range(self):
+        '''
+        returns: True if the lhs of the assignment is an array with at \
```

Comment:
I guess this is a question about what we support in the PSyIR. I was assuming (perhaps wrongly) that Fortran array notation e.g. `a = b * c` would be represented in the PSyIR using ranges (using `lbound` and `ubound`) in the same way that `a(:) = b(:) * c(:)` would. If so, this becomes an Fortran front-end problem, which I think is the correct way to deal with this. So, in answer to your last question, yes, it would return False as it would not contain a range.

Reply:
Oh yes, that makes sense.